### PR TITLE
initial version for asf proposal

### PR DESCRIPTION
**Pull request by tanzhongyibidu**
please check, thanks

### CODE REVIEW

File: brpc_proposal.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+ build an open community around it in the 'Apache Way'.
+
+
+## Background
+
+RPC framework is very important for Baidu's development when building high performance services.
+but the old existing ones can not fulfill our needs. so we build one from scratch in 2013.
+and many core services inside Baidu have adopted it.
+we named it as baidu rpc ,then change it as brpc for short.
+In 2017, we open source it and hope to get more adoption, and get more contribution from outside.
```
Comment (zyearn) [inline brpc_proposal.md]: 上面这段的所有首字母要改成大写吧

File: brpc_proposal.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+## Rationale
+
+brpc has been approved inside baidu, since many high performance core service are using it.
```
Comment (zyearn) [inline brpc_proposal.md]: Brpc, services

File: brpc_proposal.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+and since its open source, it has been adopted by several other companies , including Iqiyi,Didi,Sougou,BiliBili etc.
```
Comment (zyearn) [inline brpc_proposal.md]: 首字母。逗号后面是不是要加空格

File: brpc_proposal.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+## Current Status
+
+brpc has been an open source project on GitHub (https://github.com/brpc/brpc) since 2017.
```
Comment (zyearn) [inline brpc_proposal.md]: 首字母

File: brpc_proposal.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+Currently it has more than 7.3k stars, 1.6k forks, and is one of the most popular repositories in topic of rpc category in GitHub rpc catelogy.
+It has been widely used in Baidu, with 1,000,000+ instances and thousands kinds of services.
+Besides, many other companies have already used it also, such as Iqiyi, Didi, Sougou, BiliBili etc.
+
+### Meritocracy
+
+brpc was originally created by Ge Jun and Chen zhangyi inside baidu from 2013,
```
Comment (zyearn) [inline brpc_proposal.md]: 首字母
是句号

File: brpc_proposal.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+since its opensource in 2017, it already operates on meritocracy principles.
+it accepts multiple contribute from other companies.
+And now,the core developers are from several different companies.
```
Comment (zyearn) [inline brpc_proposal.md]: 首字母，逗号后空格
contribute是动词，it has received many contributions from...

File: brpc_proposal.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+### Inexperience with Open Source
+
+brpc has been an active open source project for more than one year.
+During that time, the project has attracted 30+ contributors and gained a lot of attentions.
+The core developers are all active users and followers of open source.
+
+### Homogenous Developers
+
+brpc was created inside Baidu, but after brpc was open sourced, it received a lot of bug fixes and enhancements from other developers not working at Baidu.
+and the core developers now are from different companies now.
```
Comment (zyearn) [inline brpc_proposal.md]: 首字母

File: brpc_proposal.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+Since our core developers are from different companies and many companies are using it,
+the risk of the project being abandoned is minimal.
+For example, Baidu is extensively using it in their production environment
+and many large corporations including Iqiyi, Didi, Sougou, BiliBili use it in their production applications.
+
+
+### Inexperience with Open Source
+
+brpc has been an active open source project for more than one year.
+During that time, the project has attracted 30+ contributors and gained a lot of attentions.
```
Comment (zyearn) [inline brpc_proposal.md]: attention似乎不可数。much attention

File: brpc_proposal.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+The following links provide more information about brpc in open source:
+
+Codebase at Github: https://github.com/brpc/brpc
+Issue Tracking: https://github.com/brpc/brpc/issues
+Overview: https://github.com/brpc/brpc/blob/master/docs/en/overview.md
+
+## Initial Source
+
+brpc has been developed since 2014 by a team of engineers at Baidu Inc.
```
Comment (zyearn) [inline brpc_proposal.md]: 首字母。
这里是2014，上面是2013，是不是没对上

File: brpc_proposal.md [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+We currently use Github to maintain our source code and track issues at https://github.com/brpc/brpc.
+We need to move our repository to Apache infrastructure.
+
+## Source and Intellectual Property Submission Plan
+
+brpc source code is available under Apache V2 license and owned by Baidu.
```
Comment (zyearn) [inline brpc_proposal.md]: 首字母

File: brpc_proposal.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+We will work with the committers to get ICLAs signed. We will provide a Software Grant Agreement from an authorized signer per https://www.apache.org/licenses/software-grant-template.pdf
+
+## External Dependencies
+
+brpc has the following external dependencies.
```
Comment (zyearn) [inline brpc_proposal.md]: 首字母？

File: brpc_proposal.md [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+* dev@brpc.incubator.apache.org
+* commits@brpc.incubator.apache.org
+
+### Git Repositories:
+
+Upon entering incubation, we want to transfer the existing repo from https://github.com/brpc/brpc to Apache infrastructure like https://github.com/apache/incubator-brpc.
+
+### Issue Tracking:
+
+brpc currently uses GitHub to track issues. Would like to continue to do so while we discuss migration possibilities with the ASF Infra committee.
```
Comment (zyearn) [inline brpc_proposal.md]: 首字母

File: brpc_proposal.md [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+
+## Sponsors
+
+### Champion
+
+* todo
+
+### Mentors
+
+* todo
```
Comment (zyearn) [inline brpc_proposal.md]: 这边的todo怎么填呢

File: brpc_proposal.md [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+## Abstract
+
+brpc is an industrial-grade RPC framework for building reliable and
+high-performance services.
+
+## Proposal
+
+We propose to contribute the brpc codebase and associated artifacts
+(e.g. documentation etc.) to the Apache Software Foundation, and aim to
+ build an wider open community around it in the 'Apache Way'.
```
Comment (zyearn) [inline brpc_proposal.md]: a wider

File: brpc_proposal.md [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+
+
+## Background
+
+RPC framework is very important for Baidu's development when building high performance services,
+but the old existing ones can not fulfill our needs, so we build one from scratch in 2014,
+and many core services inside Baidu have adopted it.
```
Comment (zyearn) [inline brpc_proposal.md]: 英文句子不能用逗号连接起来，要拆一下：
RPC framework is very important for Baidu's development in building high-performance services.
But the old existing ones can't meet our needs, so we built another one from scratch in 2014, which is used by many core services inside Baidu.

File: brpc_proposal.md [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+We named it as baidu rpc at first ,then change it as brpc for short.
```
Comment (zyearn) [inline brpc_proposal.md]: 1) 是不是baidu-rpc?
2) 逗号的后面是空格而不是前面？
3) change it to ...

File: brpc_proposal.md [hunk 17]
```diff
@@ -0,0 +1,167 @@
...
+In 2017, we open source it and hope to get more adoption, and get more contributions from outside.
```
Comment (zyearn) [inline brpc_proposal.md]: 一般用一个and，放在最后：In 2017, we open source it, hope to get more adoption, and get more contributions from outside.

File: brpc_proposal.md [hunk 18]
```diff
@@ -0,0 +1,167 @@
...
+
+## Rationale
+
+brpc has been approved inside baidu, since many high performance core services are using it.
+And since its open source, it has been adopted by several other companies , including Iqiyi, Didi, Sougou, BiliBili etc.
```
Comment (zyearn) [inline brpc_proposal.md]: 逗号前面的空格放到后面？

File: brpc_proposal.md [hunk 19]
```diff
@@ -0,0 +1,167 @@
...
+brpc has been an open source project on GitHub (https://github.com/brpc/brpc) since 2017.
+
+Currently it has more than 7.3k stars, 1.6k forks, and is one of the most popular repositories in topic of rpc category in GitHub rpc catelogy.
+It has been widely used in Baidu, with 1,000,000+ instances and thousands kinds of services.
+Besides, many other companies have already used it also, such as Iqiyi, Didi, Sougou, BiliBili etc.
+
+### Meritocracy
+
+brpc was originally created by Ge Jun and Chen zhangyi inside baidu from 2014.
+Since its opensource in 2017, it already operates on meritocracy principles.
```
Comment (zyearn) [inline brpc_proposal.md]: it has already followed meritocracy principles是不是更准确些？翻译为遵循原则

File: brpc_proposal.md [hunk 20]
```diff
@@ -0,0 +1,167 @@
...
+It accepts multiple contribute from other companies.
```
Comment (zyearn) [inline brpc_proposal.md]: 这里contribute是动词

File: brpc_proposal.md [hunk 21]
```diff
@@ -0,0 +1,167 @@
...
+And now,the core developers are from several different companies.
```
Comment (zyearn) [inline brpc_proposal.md]: 逗号后面空格？

File: brpc_proposal.md [hunk 22]
```diff
@@ -0,0 +1,167 @@
...
+### Homogenous Developers
+
+brpc was created inside Baidu, but after brpc was open sourced, it received a lot of bug fixes and enhancements from other developers not working at Baidu.
+And the core developers now are from different companies now.
+
+### Reliance on Salaried Developers
+
+Baidu invested in brpc as a general rpc framework used in company widely.
+The core developers have been dedicated to this project for about four years.
+And after its opensource, developers around the world have involved in.
```
Comment (zyearn) [inline brpc_proposal.md]: 这里用了opensource，但很多其他需要名词的地方用了open source，要统一一下？

File: brpc_proposal.md [hunk 23]
```diff
@@ -0,0 +1,167 @@
...
+Besides, we want more developers and researchers to contribute to the project.
+
+### An Excessive Fascination with the Apache Brand
+
+The mission of brpc is to help developers build reliable and high-performance services quickly and easily.
+It has been widely used in production environment throughout Baidu and after opensource, it has gained much attention and attracted developers all over the world.
+Apache Brand is very respected. We are very honored to have the opportunity to join ASF, with the understanding that its brand policies being respected,
```
Comment (zyearn) [inline brpc_proposal.md]: 最后的逗号是句号

File: brpc_proposal.md [hunk 24]
```diff
@@ -0,0 +1,167 @@
+# brpc Proposal
+
+## Abstract
+
+brpc is an industrial-grade RPC framework for building reliable and
+high-performance services.
```
Comment (jamesge) [inline brpc_proposal.md]: 此处的换行与下面比较长的行不一致。

File: brpc_proposal.md [hunk 25]
```diff
@@ -0,0 +1,167 @@
...
+ build a wider open community around it in the 'Apache Way'.
+
+
+## Background
+
+RPC framework is very important for Baidu's development when building high performance services.
+But the old existing ones can not fulfill our needs, so we build one from scratch in 2014,
+and many core services inside Baidu have adopted it.
+We named it as baidu-rpc at first, then change its name as brpc for short.
+And in 2017, we open source it and hope to get more adoption, and get more contributions from outside.
```
Comment (jamesge) [inline brpc_proposal.md]: The RPC framework used in Baidu before 2014 was developed at 2008 and limited in protocols and performance, and there were also serveral implementations focused on their own scenarios from Baidu's different BU. As an infrastructural team in Baidu, we tried to build a new framework to unify all RPC scenarios inside. The framework was named "baidu-rpc" internally the early versions were adopted and online at late 2014.  The framework was rapidly iterated at 2015-2017, and thousands kinds of services and almost all core services adopted it. And in 2017, we opensourced it as "brpc" and hope to get more adoptions and contributions from outside. At the time of opensourcing, there're more than 1 million instances inside Baidu using baidu-rpc (not counting clients).

Comment (tanzhongyibidu) [reply brpc_proposal.md]: ok, I will use this one

**Pull request closed without merging**